core/src/main/kotlin/api/ProductsRoute.kt

```diff
@@ -398,14 +398,23 @@ fun Route.products() = route("products/{productId}") {
             val productId = call.requireIdParameter("productId")
             val createOrtRun = call.receive<CreateOrtRun>()
 
+            requireNotNull(createOrtRun.jobConfigs) { "jobConfigs must be provided" }
+
             val userDisplayName = call.principal<OrtPrincipal>()?.let { principal ->
                 UserDisplayName(principal.getUserId(), principal.getUsername(), principal.getFullName())
             }
 
-            val repositoryIds = productService.getRepositoryIdsForProduct(productId)
+            val repositoryIds = createOrtRun.repositoryIds?.let { providedIds ->
+                val productRepoIds = productService.getRepositoryIdsForProduct(productId)
+                providedIds.filter { it in productRepoIds }.takeIf { it.isNotEmpty() }
+                    ?:
+                    throw IllegalArgumentException("None of the provided repository IDs belong to product $productId")
+            } ?: productService.getRepositoryIdsForProduct(productId)
+
+            println("Triggering runs for repositories: $repositoryIds with request: $createOrtRun")
```

Comment:
Please don't use any `println` in library code (as it cannot be disabled by the application that uses the library). Use logging instead if needed.